modules/core/src/main/java/org/locationtech/jts/geom/impl/CoordinateArraySequence.java

```diff
@@ -75,8 +75,11 @@ public CoordinateArraySequence(Coordinate[] coordinates, int dimension) {
   
   /**
    * Constructs a sequence based on the given array 
-   * of {@link Coordinate}s (the
-   * array is not copied).
+   * of {@link Coordinate}s (the array is not copied).
+   * <p>
+   * Ensure the array is of the indicated dimension and measures. The method
```

Comment:
Hmm... I wonder if this explanation should be expanded.  

The word "Ensure" is throwing me here.  Read quickly, could one think that the method is ensuring consistency?  (Like it was previously?)  

Also, if the requirement is on the inputs, it seems like it'd be better on the description of the coordinates variable.

Honestly, mixing Coordinates and using the common ordinates ought to be fine.  Right?

As a concrete suggestion how about something this:

"CoordinateArray should likely contain consistent implementations of the Coordinate for most uses.  Mixing types may lead to issues.  Callers can create such consistent arrays using {@link CoordinateArrays#enforceConsistency(Coordinate[])}."